code/unit_test/test_waveEqn.py

```diff
@@ -0,0 +1,319 @@
+										- 0.2040293534683072])
+
+	basis_array_analytical[3] = np.array([11.38981374849497,\
+										- 2.383879109609436,\
+										- 24.03296250200938,\
+										5.030080255538657,\
+										15.67350804691132,\
+										- 3.28045297599924,\
+										- 3.030359293396907,\
+										0.6342518300700298])
+
+	basis_array_analytical[4] = np.array([-11.38981374849497,\
+										- 2.383879109609437,\
+										24.03296250200939,\
+										5.030080255538648,\
+										- 15.67350804691132,\
+										- 3.28045297599924,\
+										3.030359293396907,\
+										0.6342518300700299])
+
+	basis_array_analytical[5] = np.array([10.35813682892759,\
+										6.128911440984293,\
+										-18.68335515838398,\
+										- 11.05494463699297,\
+										8.670037141196786,\
+										5.130062549476987,\
+										- 0.3448188117404021,\
+										- 0.2040293534683072])
+	basis_array_analytical[6] = np.array([-8.140722718246403,\
+										- 7.096594831382852,\
+										11.34747768400062,\
+										9.89205188146461, \
+										-3.331608712119162, \
+										- 2.904297073479968,\
+										0.1248537463649464,\
+										0.1088400233982081])
+	basis_array_analytical[7] = np.array([3.351562500008004,\
+										3.351562500008005, \
+										- 3.867187500010295,\
+										- 3.867187500010298,\
+										1.054687500002225, \
+										1.054687500002224, \
+										- 0.039062499999931,\
+										- 0.03906249999993102])
+				
+	basis_array_analytical = af.interop.np_to_af_array(basis_array_analytical)
+	
+	assert af.sum(af.abs(basis_array_analytical - gvar.lBasisArray)) < threshold
+
+
+def test_Integral_Li_Lp():
+	'''
+	Test function to check the A_matrix function in wave_equation module.
+	
+	Obtaining the A_matrix from the function and setting the value of
+	all elements above a certain threshold to be 1 and plotting it.
+	'''
+	threshold          = 1e-5
+	gvar.populateGlobalVariables(8, 8)
+	A_matrix_structure = np.zeros([gvar.N_LGL, gvar.N_LGL])
+	non_zero_indices  = np.where(np.array(wave_equation.A_matrix()) > threshold)
+	
+	A_matrix_structure[non_zero_indices] = 1.
+	
+	plt.gca().invert_yaxis()
+	plt.contourf(A_matrix_structure, 100, cmap = 'Blues')
+	plt.axes().set_aspect('equal')
+	plt.colorbar()
+	plt.show()
+	
+	return
+
+
+def test_A_matrix():
+	'''
+	Test function to check the A matrix obtained from wave_equation module with
+	one obtained by numerical integral solvers.
+	
+	NOTE
+	----
+	A diagonal A-matrix is used as a reference in this unit test.
+	The A matrix calculated analytically gives a different matrix.
+	
+	'''
+	threshold = 1e-8
+	gvar.populateGlobalVariables(8, 8)
+	
+	reference_A_matrix = af.tile(gvar.lobatto_weights, 1, gvar.N_LGL)\
+		* af.identity(gvar.N_LGL, gvar.N_LGL, dtype = af.Dtype.f64)
+	
+	test_A_matrix = wave_equation.A_matrix()
+	error_array   = af.abs(reference_A_matrix - test_A_matrix)
+	
+	assert af.algorithm.max(error_array) < threshold
+
+
+def test_dLp_xi():
+	'''
```

Comment:
* Since we are now having separate sagews for different tests. We should also reference
the sage worksheet here. We will have to make the sage worksheet public. I will have to
discuss whether we should make the sage worksheet public or not.